### PR TITLE
OSDOCS-3969 Alibaba VPC Installation

### PR DESCRIPTION
**Pull request by GroceryBoyJr**
Status: **Work In Progress. Do not review, do not merge.**. This is a restored PR based on a prior PR. I have already gotten one peer review but that PR had to be closed. This one is built from restored files of the last PR.

I have SME LGTM, and have requested QE LGTM.

Problem: This PR adds a new section documenting Alibaba VPC installation

Version(s): 4.10+

Issue:
How to install OpenShift into an Alibaba VPC https://issues.redhat.com/browse/OSDOCS-3969

FORMER PR for this issue: 
https://github.com/openshift/openshift-docs/pull/44593

Link to docs preview:
https://54312--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_alibaba/installing-alibaba-vpc.html
Alternately, there's also a local build preview at:
http://file.rdu.redhat.com/jbrigman/alibaba-vpc/installing/installing_alibaba/installing-alibaba-vpc.html
(These are duplicate previews so you only need to use one for reviewing.)

Additional information:
Gaurav Singh, PM. QE: Jianli Wei, SME: Thomas Wiest (Thomas has left the company but provided LGTM before departure and as-of the last QE update from Jianli Wei.)

### CODE REVIEW

**Comment by GroceryBoyJr** [comment]
Hello @jianli-wei . I had to close the prior ALIBABA-VPC PR and I have reopened this one using files restored from the prior PR. Please, may I have your QE review and LGTM if it looks OK to you?

File: modules/installation-configuration-parameters.adoc
Comment (jianli-wei) [inline modules/installation-configuration-parameters.adoc:357]: FYI The default is `OVNKubernetes` with OCP 4.12.
```
$ ./openshift-install version
./openshift-install 4.12.0-0.nightly-2023-01-08-142418
built from commit 9019444f1354a63991e2bc058da9539ac21061f6
release image registry.ci.openshift.org/ocp/release@sha256:0b0a195b0360934e726b42f302a9023593234f2545ef0cc1e03b294627e42fbc
release architecture amd64
$ ./openshift-install explain installconfig.networking.networkType
KIND:     InstallConfig
VERSION:  v1

RESOURCE: <string>
  Default: "OVNKubernetes"
  NetworkType is the type of network to install. The default value is OVNKubernetes.

$

File: modules/installation-configuration-parameters.adoc
Comment (jianli-wei) [inline modules/installation-configuration-parameters.adoc:1668]: It seems wrong spelling on "...The **provate** zone must be associated...".

File: modules/installation-configuration-parameters.adoc
Comment (jianli-wei) [inline modules/installation-configuration-parameters.adoc:1712]: Duplicate with line 1667 - 1668? Please confirm and keep one only, thanks!

**Review by jianli-wei — CHANGES_REQUESTED** [review]

**Comment by GroceryBoyJr** [comment]
/remove-label peer-review-needed
Removing from peer review queue to make QE requested edits.

File: modules/installation-configuration-parameters.adoc
Comment (GroceryBoyJr) [reply modules/installation-configuration-parameters.adoc:357]: Hello @jianli-wei - thank you for pointing out that important change in this module. That update is not going to be made in my PR, we have a team member who is tasked with changing that in the entire documentation set. I have been asked not to make that edit and to leave it alone so it can be changed automatically.

**Comment by GroceryBoyJr** [comment]
@jianli-wei PTAL, seek your LGTM, please.

File: modules/installation-custom-alibaba-vpc.adoc
Comment (bscott-rh) [inline modules/installation-custom-alibaba-vpc.adoc:7]:
```suggestion
= Using a custom VPC
```
Minimalism suggestion

File: modules/installation-custom-alibaba-vpc.adoc
Comment (bscott-rh) [inline modules/installation-custom-alibaba-vpc.adoc:9]:
```suggestion
In {product-title} {product-version}, you can deploy a cluster into existing subnets in an existing Virtual Private Cloud (VPC) in the Alibaba Cloud Platform. By deploying {product-title} into an existing Alibaba VPC, you can avoid limit constraints in new accounts and more easily adhere to your organization's operational constraints. If you cannot obtain the infrastructure creation permissions that are required to create the VPC, use this installation option. You must configure networking using vSwitches.
```
Rewording/minimalism suggestion

File: modules/installation-custom-alibaba-vpc.adoc
Comment (bscott-rh) [inline modules/installation-custom-alibaba-vpc.adoc:32]:
```suggestion
* The vSwitches' CIDRs belong to the machine CIDR that you specified.
```
Nit in case there are multiple vSwitches

File: installing/installing_alibaba/installing-alibaba-vpc.adoc
Comment (bscott-rh) [inline installing/installing_alibaba/installing-alibaba-vpc.adoc:61]: Duplicate line can be removed

**Review by bscott-rh — CHANGES_REQUESTED** [review]
Left a few wording/minimalism suggestions. I believe the modules/installation-configuration-parameters.adoc file is out of date and needs to be rebased and updated with the alibaba VPC changes.

File: modules/cluster-entitlements.adoc
Comment (jianli-wei) [reply modules/cluster-entitlements.adoc:126]: FYI to install on Alibabacloud in a restricted/disconnected network would fail due to https://issues.redhat.com/browse/OCPBUGS-2388.

File: modules/cco-ccoctl-creating-at-once.adoc
Comment (jeana-redhat) [reply modules/cco-ccoctl-creating-at-once.adoc:99]: Thanks Ben - @GroceryBoyJr Not sure why you have this removed in your PR, but feel free to ping me to chat about this if you need :smiley:

Comment (GroceryBoyJr) [reply modules/cco-ccoctl-creating-at-once.adoc:99]: @jeana-redhat I was using too-old of a version of the file and for some reason I wasn't getting the updated file in my rebases. I have corrected, this, thanks Ben for flagging this!

File: installing/installing_alibaba/installing-alibaba-vpc.adoc
Comment (GroceryBoyJr) [reply installing/installing_alibaba/installing-alibaba-vpc.adoc:61]: Fixed, thank you Ben!

**Comment by GroceryBoyJr** [comment]
This PR is on hold for now pending review by @GroceryBoyJr @mjpytlak and @bscott-rh

**Comment by GroceryBoyJr** [comment]
Hello @jianli-wei , may I have your final comment or LGTM please?

File: modules/installation-configuration-parameters.adoc
Comment (jianli-wei) [inline modules/installation-configuration-parameters.adoc:1652]: Suggest to replace `ecs.g6.large` with `ecs.g6.xlarge`, as control-plane nodes requires 4 vCPU and 16G virtual RAM as a minimum. 

FYI
https://www.alibabacloud.com/help/en/elastic-compute-service/latest/instance-family#g6
https://docs.openshift.com/container-platform/4.12/installing/installing_gcp/installing-gcp-customizations.html#installation-minimum-resource-requirements_installing-gcp-customizations

**Review by jianli-wei — CHANGES_REQUESTED** [review]
Besides, another 2 points, thanks! 

In "Sample customized install-config.yaml file for Alibaba Cloud", it seems you removed `vpcID` & `vswitchIDs` section. Suggest to add it back. 

In "Creating credentials for OpenShift Container Platform components with the ccoctl tool", if possible, suggest to tell a link on how to get `ccoctl` (e.g. https://docs.openshift.com/container-platform/4.12/authentication/managing_cloud_provider_credentials/cco-mode-gcp-workload-identity.html#cco-ccoctl-configuring_cco-mode-gcp-workload-identity).

**Comment by jianli-wei** [comment]
> Hello @jianli-wei , may I have your final comment or LGTM please?

@GroceryBoyJr Except the above comment, I have no other suggestion. Thanks!

File: modules/cco-ccoctl-creating-at-once.adoc
Comment (GroceryBoyJr) [reply modules/cco-ccoctl-creating-at-once.adoc]: This issue is resolved by manually pulling an updated cco-ccoctl-creating-at-once.adoc file and re-editing my changes.

File: modules/cluster-entitlements.adoc
Comment (GroceryBoyJr) [reply modules/cluster-entitlements.adoc:126]: @bscott-rh I have added these lines back. @jianli-wei I will keep an eye on OCPBUGS-2388, thank you sir!

File: modules/installation-alibaba-config-yaml.adoc
Comment (GroceryBoyJr) [reply modules/installation-alibaba-config-yaml.adoc]: This is done, thank you @bscott-rh !

File: modules/installation-alibaba-config-yaml.adoc
Comment (GroceryBoyJr) [reply modules/installation-alibaba-config-yaml.adoc]: I have changed these callouts to sentences, great catch, @bscott-rh !

File: modules/installation-custom-alibaba-vpc.adoc
Comment (GroceryBoyJr) [reply modules/installation-custom-alibaba-vpc.adoc:9]: Damn good suggestion, @bscott-rh. Much appreciate your feedback!

File: modules/installation-custom-alibaba-vpc.adoc
Comment (GroceryBoyJr) [reply modules/installation-custom-alibaba-vpc.adoc:32]: Change made, although...I need to check the style guide on that trailing apostrophe. I'm not sure we need that?

File: modules/installation-configuration-parameters.adoc
Comment (GroceryBoyJr) [reply modules/installation-configuration-parameters.adoc:1652]: Thank you so much! This change is made, @jianli-wei

File: modules/installation-custom-alibaba-vpc.adoc
Comment (bscott-rh) [reply modules/installation-custom-alibaba-vpc.adoc:32]: I'm not sure either.. I was thinking of it in plural possessive, "The CIDRs of the vSwitches" aka "The vSwitches' CIDRs" ... may be best to just reword as "The CIDR of each vSwitch belongs to..." or some such. Up to you

**Comment by mjpytlak** [comment]
@GroceryBoyJr Your Alibaba VPC docs merged a while back -- cannot reference the exact PR at the moment. Can this PR be closed?

**Pull request closed without merging**
